### PR TITLE
Add to_tensor method and update evaluator

### PR DESCRIPTION
**Pull request by vfdev-5**
This covers possible issues using `create_supervised_evaluator` if targets are not simple `torch.Tensor`

### CODE REVIEW

**Comment by jasonkriss** [comment]
@vfdev-5 Do you mind giving a concrete use case where the supervised targets are not tensors?

**Comment by vfdev-5** [comment]
@jasonkriss yep, in a detection task with SSD targets are encoded  boxes and labels (location, cls). See [this](https://github.com/kuangliu/torchcv/blob/master/torchcv/loss/ssd_loss.py#L35)

**Comment by vfdev-5** [comment]
@alykhantejani what do you think about this ?

File: ignite/engines/evaluator.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def _inference(batch):
         model.eval()
         x, y = _prepare_batch(batch)
         y_pred = model(x)
-        return y_pred.data.cpu(), y.data.cpu()
+        return to_tensor(y_pred, cpu=not cuda), to_tensor(y, cpu=not cuda)
```
Comment (alykhantejani) [inline ignite/engines/evaluator.py]: nit: cpu could always be true here. I think its a no-op if it's already on the cpu

Comment (vfdev-5) [reply ignite/engines/evaluator.py]: Do you think it would not impact metrics computation time if it is on cpu ?

**Comment by jasonkriss** [comment]
Thanks @vfdev-5! Seems totally reasonable to me.

**Comment by alykhantejani** [comment]
Thanks @vfdev-5!
